pkg/logger/ecs_error_log.go

```diff
@@ -0,0 +1,188 @@
+package logger
+
+import (
+	"fmt"
+	"go.uber.org/zap/zapcore"
+	"log"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+type ECSErrorLogger struct {
+	ErrorFilePath    string
+	ErrorLogMaxAge   int
+	ErrorFileMaxSize int
+
+	queue     []zapcore.Entry
+	ticker    time.Ticker
+	errorFile *os.File
+	size      int
+}
+
+// ECSErrorReporter func help select the different case between either write error log into file or
+// rotate and delete old logs in file based on TTL
+func (l *ECSErrorLogger) ECSErrorReporter() {
+	defer l.ticker.Stop()
+	for {
+		select {
+		// when new error add in the channel, this case will be selected and write it into error log file
+		case e := <-errorLogChannel:
+			l.Write(e)
+		// this case will run every x mins (x is configurable) to delete the error which already expired, error log max age
+		// depends on STATUS_MESSAGE_TTL
+		case <-l.ticker.C:
+			l.processTimeout()
+		}
+	}
+}
+
+func (l *ECSErrorLogger) Write(newError zapcore.Entry) {
+	if l.errorFile == nil {
+		if err := l.openExistingOrNewErrorFile(); err != nil {
+			log.Printf("could not open error log file when write new error, err: %v", err)
+		}
+	}
+	newErrorLog := []byte(fmt.Sprintf("{%+v, Level:%+v, Caller:%+v, Message:%+v}\r\n",
+		newError.Time, newError.Level, newError.Caller, newError.Message))
+
+	if len(newErrorLog) > l.ErrorFileMaxSize {
+		log.Printf("error size exceed the max size of error file")
```

Comment:
In this and the other error print statements, could we add something to make it clear that these come from the error reporter?

May be something like:


```
log.Printf("[ecs error reporter] error size exceed the max size of error file")
```

Reply:
okay, will change like that